src/org/traccar/database/DeviceManager.java

```diff
@@ -72,158 +68,125 @@ public DeviceManager(DataManager dataManager) {
     }
 
     private void updateDeviceCache(boolean force) throws SQLException {
-        boolean needWrite;
-        devicesLock.readLock().lock();
-        try {
-            needWrite = force || System.currentTimeMillis() - devicesLastUpdate > dataRefreshDelay;
-        } finally {
-            devicesLock.readLock().unlock();
-        }
-
+        boolean needWrite = force || System.currentTimeMillis() - devicesLastUpdate.get() > dataRefreshDelay;
```

Comment:
I don't think it's the right way to use atomics. I should it's better to use `compareAndSet`. What do you think?